scripts/vehicle_node.py

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env python
+import rospy
+import tf
+from math import ceil
+import numpy as np
+import copy
+import threading
+import sensor_msgs.point_cloud2 as pcl2
+from scipy.ndimage import gaussian_filter
+from numpy import sin, cos
+from geometry_msgs.msg import Twist, PoseStamped, Pose, Point
+from std_msgs.msg import Float32MultiArray, MultiArrayDimension, Header
+from sensor_msgs.msg import PointCloud2, JointState
+from scipy.interpolate import griddata
+from visualization_msgs.msg import Marker
+
+
+
+class VehicleBot:
+
+    def __init__(self):
+
+        rospy.init_node('vehicle_node', anonymous=True)
+        
+        #Load parameters to be used in node
+        self.mu = rospy.get_param('/mu')
+        self.sigma = rospy.get_param('/sigma')
+        self.sigma_filt = rospy.get_param('/sigma_filt')
+        self.height = rospy.get_param('/height')
+        self.width = rospy.get_param('/width')
+        self.resolution =  rospy.get_param('/resolution')
```

Comment:
As defined, your parameterization doesn't easily support non-square environments.  You end up stretching out each of your terrain cells.  I would suggest redefining resolution to be the size of a single terrain grid cell.